src/main/kotlin/no/skatteetaten/aurora/mokey/service/StorageGridObjectAreaService.kt

```diff
@@ -1,32 +1,141 @@
 package no.skatteetaten.aurora.mokey.service
 
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.TimeoutCancellationException
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import mu.KotlinLogging
+import no.skatteetaten.aurora.mokey.model.StorageGridObjectArea
 import no.skatteetaten.aurora.mokey.model.StorageGridObjectAreaDetails
+import org.apache.commons.lang3.exception.ExceptionUtils
 import org.springframework.beans.factory.annotation.Value
+import org.springframework.scheduling.annotation.Scheduled
 import org.springframework.stereotype.Service
+import org.springframework.util.StopWatch
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import java.time.Duration
+import java.util.concurrent.ConcurrentHashMap
+
+private val logger = KotlinLogging.logger {}
 
 @Service
 class StorageGridObjectAreaService(
     val userClient: OpenShiftUserClient,
+    val applicationDataService: ApplicationDataServiceOpenShift,
+    val openShiftServiceAccountClient: OpenShiftServiceAccountClient,
     @Value("\${openshift.cluster}") val cluster: String,
+    @Value("\${mokey.crawler.timeout:3m}") val crawlerTimeout: Duration,
+    @Value("\${mokey.cache.affiliations:}") val affiliationsConfig: String,
 ) {
+    val affiliations: List<String>
+        get() = if (affiliationsConfig.isBlank()) emptyList()
+        else affiliationsConfig.split(",").map { it.trim() }
+    val cache = ConcurrentHashMap<String, List<StorageGridObjectArea>>()
+
+    suspend fun getAllStorageGridObjectAreasForAffiliationFromCache(affiliation: String): List<StorageGridObjectAreaDetails> {
```

Comment:
Burde vi heller ha dette i "hoved" cachen? 

Jeg er redd for at dette kan bli tungt for mokey. 

@stromland føler jeg er best til å svare på dette